service/src/main/java/bio/terra/workspace/db/WorkspaceDao.java

```diff
@@ -84,6 +85,7 @@ public UUID createWorkspace(Workspace workspace) {
             + " cast(:properties AS jsonb), :workspace_stage)";
 
     final String workspaceUuid = workspace.getWorkspaceId().toString();
+    ControllerValidationUtils.validateUserFacingId(workspace.getUserFacingId());
```

Comment:
I think this makes sense, but I don't know if we intended this method to be called from below the controller layer. @zloery do you think it matters?

Reply:
I checked with @ddietterich, he's ok with this. I'm calling this in relevant controller places. I also call in Dao in case there's some controller place I missed, or some new controller in the future. If my first PR called in dao, this bug (PF-1616) wouldn't have happened.

Added comment since it's strange to see `ControllerValidationUtils` here, thanks.

Reply:
strictly speaking if we checked every endpoint where a user-facing ID could be generated/passed in we shouldn't need this, but it's a static validation method so it doesn't really cost much. 

+1 we should be calling it from all controller endpoints where user facing IDs could come in/be generated/be modified, but also having it in the DAO with a comment sgtm